### PR TITLE
Add submit task mutation

### PR DESCRIPTION
**Pull request by ConnorBoulais**
This PR integrates the `submitTask` mutation with the submit button on the task screen, so that you can mark a task as done by clicking submit.

This PR allows fixes Defect HYD-127 which was causing some answers to appear like they were not being saved because they were being overridden by the cache.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -71,9 +71,11 @@ function App() {
     if(graphQLErrors){
       graphQLErrors.forEach(({ message, locations, path }) => {
         console.warn(message);
-      })  
-      hist.push({
-        pathname:'/',
+        if(message.includes('JWT')){
```
Comment (Bart-Ilg) [inline src/App.js:74]: good to add condition

File: src/Components/FRQuestion.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
```
Comment (Bart-Ilg) [inline src/Components/FRQuestion.js:1]: good to clear unneeded imports

File: src/Components/Task.js [hunk 3]
```diff
@@ -122,13 +124,26 @@ function Task(props) {
...
+    }).then((response) => {
+      console.log(response);
+      alert(`${title} submitted.`);
+      props.history.push({
+        pathname: "/mission",
+        state: {
+          id: data.task.missionId
+        }
+      });
+    }).catch((error) => {
```
Comment (Bart-Ilg) [inline src/Components/Task.js:141]: good to catch and log error

**Review by ryanmadamba — APPROVED** [review]
Looks good. This will be helpful to fix the defect that was reported.

File: src/Components/Task.js [hunk 4]
```diff
@@ -122,13 +124,26 @@ function Task(props) {
...
-      pathname: "/mission",
-      state: {
-        id: data.task.missionId
+
+    submitTaskMutation({
+      variables: {
+        taskId: taskId
       }
-    });
+    }).then((response) => {
```
Comment (Bart-Ilg) [inline src/Components/Task.js:132]: looks good

File: src/gqlQueries.js [hunk 5]
```diff
@@ -279,3 +279,26 @@ export const GET_TASK_PROGRESS = gql`
   }
 `;
  
+export const SUBMIT_TASK = gql`
```
Comment (Bart-Ilg) [inline src/gqlQueries.js:282]: well organized

**Review by Bart-Ilg — APPROVED** [review]
Looks good, thanks for implementing it for us

**Pull request merged**